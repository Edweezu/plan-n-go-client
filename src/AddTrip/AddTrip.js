import React from 'react';
// import { Link } from 'react-router-dom'
import config from '../config'
import TokenService from '../services/token-service'
import TripsContext from '../TripsContext';

export default class AddTrip extends React.Component {

    static contextType = TripsContext

    state = {
        error: null
    }

    handleAddTrip = (e) => {
        e.preventDefault()

        const { city, trip_name, start_date, end_date } = e.target

        const newTrip = {
            city: city.value,
            trip_name: trip_name.value,
            start_date: start_date.value,
            end_date: end_date.value,
        }

        console.log('new trippp', newTrip)
        
        return fetch(`${config.API_ENDPOINT}/trips`, {
            method: 'POST',
            headers: {
                'content-type': 'application/json',
                'authorization': `bearer ${TokenService.getAuthToken()}`
            },
            body: JSON.stringify(newTrip)
        })
        .then(res => {
            if (!res.ok) {
                return res.json().then(e => Promise.reject(e))
            }
            return res.json()
        })
        .then(trip => {
            this.context.addTrip(trip)
            this.props.history.push('/dashboard')
        })
        .catch(error => {
            this.setState({
                error
            })
        })
    }

    handleClickCancel = () => {
        this.props.history.push('/dashboard')
    }

    render () {
        return (
            <main className='add-trip'>
                <h2 className='trip-name-header'>Add a Trip</h2>
                <form className='main-form' onSubmit={this.handleAddTrip}>
                    <div className='form-flex-container'>   
                        <div className='add-trip-element'>
                            <label htmlFor='city'>
                            City
                            </label>
                            <span className='astrik'>
                                *
                            </span>
                            <input className='form-input add-trip-input' id='city' name='city' type='text' required/>
                        </div>
                        <div className='add-trip-element'>
                            <label htmlFor='trip-name'>
                                Trip Name
                            </label>
                            <span className='astrik'>
                                *
                            </span>
                            <input className='form-input add-trip-input' id='trip_name' name='trip_name' type='text' required/>
                        </div>
                    </div>
                    <div className='form-flex-container'>
                        <div className='add-trip-element'>
                            <label htmlFor='start-date'>
                                Start Date  
                            </label>
                            <span className='astrik'>
                                *
                            </span>
                            <input className='form-input add-trip-input' id='start_date' name='start_date' type='date' required/>
                        </div>
                        <div className='add-trip-element'>
                            <label htmlFor='end-date'>
                                End Date 
                            </label>
                            <span className='astrik'>
                                *
                            </span>
                            <input className='form-input add-trip-input' id='end_date' name='end_date' type='date' required/>
                        </div>
                    </div>
                    <div className='button-container'>
                        <button className='submit-button' type='submit'>
                            Submit
                        </button>
                        <button className='cancel-button' onClick={this.handleClickCancel}>
                            Cancel
                        </button>
                    </div>
                </form>
            </main>
        )
    }
}