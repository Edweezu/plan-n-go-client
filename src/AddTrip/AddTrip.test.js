import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import AddTrip from './AddTrip'


describe('AddTrip component', () => {
    it('renders the given component', () => {
        const wrapper = shallow(<AddTrip />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})