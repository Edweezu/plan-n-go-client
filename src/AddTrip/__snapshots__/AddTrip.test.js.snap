// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTrip component renders the given component 1`] = `
<main
  className="add-trip"
>
  <h2
    className="trip-name-header"
  >
    Add a Trip
  </h2>
  <form
    className="main-form"
    onSubmit={[Function]}
  >
    <div
      className="form-flex-container"
    >
      <div
        className="add-trip-element"
      >
        <label
          htmlFor="city"
        >
          City
        </label>
        <span
          className="astrik"
        >
          *
        </span>
        <input
          className="form-input add-trip-input"
          id="city"
          name="city"
          required={true}
          type="text"
        />
      </div>
      <div
        className="add-trip-element"
      >
        <label
          htmlFor="trip-name"
        >
          Trip Name
        </label>
        <span
          className="astrik"
        >
          *
        </span>
        <input
          className="form-input add-trip-input"
          id="trip_name"
          name="trip_name"
          required={true}
          type="text"
        />
      </div>
    </div>
    <div
      className="form-flex-container"
    >
      <div
        className="add-trip-element"
      >
        <label
          htmlFor="start-date"
        >
          Start Date
        </label>
        <span
          className="astrik"
        >
          *
        </span>
        <input
          className="form-input add-trip-input"
          id="start_date"
          name="start_date"
          required={true}
          type="date"
        />
      </div>
      <div
        className="add-trip-element"
      >
        <label
          htmlFor="end-date"
        >
          End Date
        </label>
        <span
          className="astrik"
        >
          *
        </span>
        <input
          className="form-input add-trip-input"
          id="end_date"
          name="end_date"
          required={true}
          type="date"
        />
      </div>
    </div>
    <div
      className="button-container"
    >
      <button
        className="submit-button"
        type="submit"
      >
        Submit
      </button>
      <button
        className="cancel-button"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </form>
</main>
`;
